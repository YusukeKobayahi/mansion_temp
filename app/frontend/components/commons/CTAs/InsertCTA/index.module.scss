@import "~/styles/variables";
@import "~/styles/utils";

.container {
  margin: 20px auto;
  padding: 20px 0;
  width: 100%;
  background-color: #f8f6f0;
}

.description {
  text-align: center;
  line-height: 1.5;
  margin-bottom: 15px;
  font-weight: bold;
  color: #746744;
  font-size: 16px;

  @include screenMd {
    padding: 0 16px;
    font-size: 12px;
  }
}

.emphasisText {
  font-weight: bold;
}

.btn {
  display: block;
  margin: 0 auto;
  padding: 16px 60px;
  background-color: #f15e3a;
  border-radius: 40px;
  font-size: 16px;
  color: #ffffff;
  cursor: pointer;
  font-weight: bold;
  position: relative;

  @include screenMd {
    box-sizing: border-box;
    padding: 16px 30px;
    margin: 0 16px;
    width: calc(100% - 32px);
  }

  &::after {
    content: "";
    position: absolute;
    top: 50%;
    right: 30px;
    width: 10px;
    height: 10px;
    border-top: 2px solid #ffffff;
    border-right: 2px solid #ffffff;
    transform: translateY(-50%) rotate(45deg);

    @include screenMd {
      right: 20px;
    }
  }
}
