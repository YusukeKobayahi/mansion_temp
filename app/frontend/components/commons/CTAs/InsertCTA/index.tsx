import styles from "./index.module.scss";
import { clickMembers } from "~/lib/utils";

const InsertCTA: React.FC = () => {
  const clickCTA = () => {
    clickMembers({
      action: "msl_cta_click",
      category: "msl_ichiran",
      label: "msl_lp_center",
      value: 1,
    });
  };
  return (
    <section className={styles.container}>
      <p className={styles.description}>
        ＼ ここには掲載していない未公開物件があります！ ／
      </p>
      <button className={styles.btn} onClick={clickCTA}>
        会員登録してオファー限定情報を手に入れる
      </button>
    </section>
  );
};

export default InsertCTA;
