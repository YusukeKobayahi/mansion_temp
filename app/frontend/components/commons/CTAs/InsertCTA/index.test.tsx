import { render } from "@testing-library/react";
import InsertCTA from ".";

describe("components/commons/CTAs/InsertCTA", () => {
  it("正常に表示される", () => {
    const { asFragment } = render(<InsertCTA />);
    expect(asFragment()).toMatchSnapshot();
  });
});
