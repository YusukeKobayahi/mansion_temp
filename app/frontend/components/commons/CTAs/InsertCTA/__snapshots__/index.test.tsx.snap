// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/commons/CTAs/InsertCTA 正常に表示される 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <p
      class="description"
    >
      ＼ ここには掲載していない未公開物件があります！ ／
    </p>
    <button
      class="btn"
    >
      会員登録してオファー限定情報を手に入れる
    </button>
  </section>
</DocumentFragment>
`;
